core/src/main/java/de/bwaldvogel/mongo/oplog/CollectionBackedOplog.java

```diff
@@ -86,4 +109,61 @@ private boolean isOplogCollection(String namespace) {
         return collection.getFullName().equals(namespace);
     }
 
+    private Document getFullDocument(Document changeStreamDocument, Document document, OperationType operationType) {
+        switch (operationType) {
+            case INSERT:
+                return (Document) document.get("o");
+            case DELETE:
+                return null;
+            case UPDATE:
+                return lookUpUpdateDocument(changeStreamDocument, document);
+        }
+        throw new IllegalArgumentException("Invalid operation type");
+    }
+
+    private Document lookUpUpdateDocument(Document changeStreamDocument, Document document) {
+        if (changeStreamDocument.containsKey("fullDocument") && changeStreamDocument.get("fullDocument").equals("updateLookup")) {
```

Comment:
minor, could refer to the static constants from Mongo Java Driver e.g. `FullDocument#UPDATE_LOOKUP`

Reply:
I'm not so sure about adding a dependency on the client java driver within the server code. I don't see the harm in it but we haven't done it so far, in the server we have our own Document class for instance, which could have been taken from the driver.

Reply:
ah yes, good point. let's leave as is.